src/cirrus/lib2/events.py

```diff
@@ -0,0 +1,149 @@
+import json
+import os
+from contextlib import AbstractContextManager, contextmanager
+from logging import Logger, getLogger
+from typing import Dict
+
+import boto3
+
+from .enums import StateEnum
+from .statedb import StateDB
+from .utils import SNSPublisher
+
+logger = getLogger(__name__)
+
+
+class WorkflowEventManager:
+    """A class for managing payload state change events, including storage of
+    state(dynamo), activity(timestream), and notifications of decisions and/or status
+    changes (SNS)."""
+
+    def __init__(
+        self,
+        logger: Logger = logger,
+        boto3_session: boto3.Session = None,
+        statedb: StateDB = None,
+        batch_size: int = 10,
+    ):
+        self.logger = logger
+        self._boto3_session = boto3_session if boto3_session else boto3.Session()
+        wf_event_topic_arn = os.getenv("CIRRUS_WORKFLOW_EVENT_TOPIC_ARN", None)
+        self.event_publisher = (
+            SNSPublisher(
+                wf_event_topic_arn,
+                logger=logger,
+                boto3_session=self._boto3_session,
+                batch_size=batch_size,
+            )
+            if wf_event_topic_arn
+            else None
+        )
+        self.statedb = (
+            statedb if statedb is not None else StateDB(session=self._boto3_session)
+        )
+
+    def flush(self):
+        if self.event_publisher:
+            self.event_publisher.execute()
+
+    @classmethod
+    @contextmanager
+    def handler(
+        cls: "WorkflowEventManager", *args, **kwargs
+    ) -> AbstractContextManager["WorkflowEventManager"]:
+        wfem = cls(*args, **kwargs)
+        try:
+            yield wfem
+        finally:
+            wfem.flush()
+
+    def announce(
+        self,
+        event_type: str,
```

Comment:
I made this comment earlier, but I think the event_type could be made into an Enum instead of a string